src/aws-cpp-sdk-core/source/auth/SSOCredentialsProvider.cpp

```diff
@@ -101,6 +101,13 @@ void SSOCredentialsProvider::Reload()
     AWS_LOGSTREAM_TRACE(SSO_CREDENTIALS_PROVIDER_LOG_TAG, "Successfully retrieved credentials with AWS_ACCESS_KEY: " << result.creds.GetAWSAccessKeyId());
 
     m_credentials = result.creds;
+    if (!m_credentials.IsEmpty()) {
```

Comment:
will this actually be called if the credentials provider is called twice? given that the credentials did not expire between the calls?